sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -108,7 +108,7 @@ public ReactorConnection(String connectionId, ConnectionOptions connectionOption
                 final Mono<AmqpEndpointState> activeEndpoint = getEndpointStates()
                     .filter(state -> state == AmqpEndpointState.ACTIVE)
                     .next()
-                    .timeout(operationTimeout, Mono.error(new AmqpException(false, String.format(
+                    .timeout(operationTimeout, Mono.error(new AmqpException(true, String.format(
```

Comment:
Have we tested this where we are converting this exception into transient error ?

Reply:
Yes I tested locally. When my home computer has wifi but the router actually is disconnected from internet, this timeout may kick in and there is no retry. After this change there is retry.